drivers/i2c/Kconfig.mspm0g3xxx

```diff
@@ -24,4 +24,19 @@ config I2C_MSPM0G3XXX_TRANSFER_TIMEOUT
 	  0 means that the driver should use the K_FOREVER value,
 	  i.e. it should wait as long as necessary.
 
+config I2C_MSPM0G3XXX_WATCHDOG_TIMEOUT
+    int "I2C watchdog reset timeout (us)"
+    default 300
+    help
+      Defines the timeout in microseconds after which the watchdog will
+      reset the I2C lines and trigger a system panic.
+
+config I2C_MSPM0G3XXX_WATCHDOG_PANIC_CODE
+    int "I2C watchdog reset panic code"
+    default 0
```

Comment:
Put it at 17 or else the reason might be a code from enum k_fatal_error_reason and we won't be able to differenciate.